osm2streets/src/road.rs

```diff
@@ -443,6 +455,45 @@ impl Road {
     pub fn common_endpoint(&self, other: &Road) -> CommonEndpoint {
         CommonEndpoint::new((self.src_i, self.dst_i), (other.src_i, other.dst_i))
     }
+
+    /// This trims a polyline on both ends. Positive trim distances mean to shorten the polyline,
+    /// and negative mean to extend the polyline in a straight line by some amount. If this returns
+    /// `None`, then the two trim distances are incompatible -- the entire polyline disappears.
+    ///
+    /// TODO Maybe move this to PolyLine directly. This is a utility method currently for A/B
```

Comment:
I'm going to try using this in the A/B Street editor and operate off the same concept of two trim distances